### DISCUSSION TITLE
Upgrading from release 7.4 to 7.7

### DISCUSSION THREAD

**Discussion by NRollo** [cesanta/mongoose#1574] [Ask questions, get answers] [answered]
In release 7.4 you could use the mg_http_upload() when a file was being uploaded via a HTTP form. A callback function would then "catch" the filename of the file being uploaded.
As I understand the changes in release 7.7 this upload now has to be done as shown in the examples "file-upload" and "form-upload". How do I "catch" the filename of the file being uploaded?
Apparently the callback function is not used in release 7.7 and the header does not seem to contain the filename in the MG_EV_HTTP_CHUNK event. Does an example exist on how to upload large files (and save them) with the help of the MG_EV_HTTP_CHUNK event?

**Comment by scaprile** [comment]
Hi, if I understand you correctly, [this tutorial](https://mongoose.ws/tutorials/file-uploads/#streaming-method) shows how to do a file upload when it is larger than the available memory and a multipart-form can't be used, as in the "form-upload" example.
Note [this piece of code](https://github.com/cesanta/mongoose/blob/9e63431/examples/file-upload/main.c#L17-L25) in the callback function, this line: 
```c
LOG(LL_INFO, ("Query string: [%.*s]", (int) hm->query.len, hm->query.ptr));
```
the query string contains the filename, you can see that in the log if you run the example.

**Reply by NRollo** [reply]
@scaprile Thanks for commenting.
I actually just now found it in the Body of the first chunk as you can below:
![image](https://user-images.githubusercontent.com/55288823/171697296-b177c064-1507-4551-9143-aee6fd1ac1e8.png)

I then tried to extract it with:
`mg_http_get_var(&hm->body, "filename", name, sizeof(name));
`
But no luck so far.

**Comment by NRollo** [comment, marked as answer]
I found my missing "Content-Disposition" header (filename) in the body part of the HTTP message.
